File: frontend/shop/src/app/shared/models/product.ts
export type Product = {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  type: string;
  brand: string;
  quantityInStock: number;
  platformType: Platform;
}

export enum Platform {
  PC = 0,
  PlayStation = 1,
  Xbox = 2,
  Nintendo = 3,
  Mobile = 4
}

File: backend/Core/Entities/Product.cs
﻿using System.Text.Json.Serialization;

namespace Core.Entities;

public class Product : BaseEntity
{
    public required string Name { get; set; }
    public required string Description { get; set; }
    public decimal Price { get; set; }  
    public required string PictureUrl { get; set; } 
    public required string Type { get; set; }
    public required string Brand{ get; set; }
    public int QuantityInStock { get; set; }
    [JsonConverter(typeof(JsonStringEnumConverter))] 
    public Platform PlatformType { get; set; }
}